Social/Facebook/Oauth.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+/*
+ * This file is part of the Integrated package.
+ *
+ * (c) e-Active B.V. <integrated@e-active.nl>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Integrated\Bundle\SocialBundle\Social\Facebook;
+
+use Facebook\Exceptions\FacebookResponseException;
+use Facebook\Exceptions\FacebookSDKException;
+use Facebook\Facebook;
+use Integrated\Bundle\SocialBundle\Oauth\OauthInterface;
+use Integrated\Common\Channel\Connector\Config\OptionsInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Class Oauth
+ * @package Integrated\Bundle\SocialBundle\Social\Facebook
+ */
+class Oauth implements OauthInterface
+{
+    private $facebook_app_id;
+
+    private $facebook_app_id_secret;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * Oauth constructor.
+     * @param $facebook_app_id
+     * @param $facebook_app_id_secret
+     * @param RequestStack $requestStack
+     */
+    public function __construct($facebook_app_id, $facebook_app_id_secret, RequestStack $requestStack)
+    {
+        $this->facebook_app_id = $facebook_app_id;
+        $this->facebook_app_id_secret = $facebook_app_id_secret;
+        $this->requestStack = $requestStack->getCurrentRequest();
+    }
+
+    /**
+     * @param $connector
+     * @param $admin_url
+     * @return string
+     */
+    public function login($connector, $admin_url)
```

Comment:
I didn't find any use of the login or callback method with in this bundle so what is the use of these methods ?